athena-snowflake/src/main/java/com/amazonaws/athena/connectors/snowflake/SnowflakeMetadataHandler.java

```diff
@@ -178,6 +181,48 @@ public Schema getPartitionSchema(final String catalogName)
                 .addField(BLOCK_PARTITION_COLUMN_NAME, Types.MinorType.VARCHAR.getType());
         return schemaBuilder.build();
     }
+
+    private String getPrimaryKey(TableName tableName) throws Exception 
+    {
+        String primaryKey = "";
+        try (Connection connection = getJdbcConnectionFactory().getConnection(getCredentialProvider())) {
+            try (PreparedStatement preparedStatement = connection.prepareStatement(SHOW_PRIMARY_KEYS_QUERY + tableName.getTableName());
+                ResultSet rs = preparedStatement.executeQuery()) {
+                while (rs.next()) {
+                    // Concatenate multiple primary keys if they exist
+                    primaryKey += rs.getString(PRIMARY_KEY_COLUMN_NAME) + ", ";
+                }
+            }
+        }
+        primaryKey = primaryKey.replaceAll(", $", "");
+        if (!primaryKey.equals("") && hasUniquePrimaryKey(tableName, primaryKey)) {
```

Comment:
I'd recommend to check if Empty or Null you can use `Strings.isNullOrEmpty` from `import com.google.common.base.Strings;`

See example here; `athena-oracle/src/main/java/com/amazonaws/athena/connectors/oracle/OracleQueryStringBuilder.java`

Reply:
Fixed! Thanks